File: exo6/exo6.rb
class Exo6
    def self.perroquet()
        puts "Veuillez ecrire quelque chose\n"
        ipt = gets.chomp
        puts "Vous avez ecrit : " + ipt + "\n"
        
        actionValide = false
        while(actionValide == false)
            puts "Voulez vous quitter ? o/n : "
            action = gets.chomp
            if (action == "o" or action == "O")
                return
            elsif (action == "n" or action == "N")
                actionValide = true
            else
                print("Veuillez entrer une reponse valide\n")
            end
        end
        
        print("\n")
        perroquet()
    end

    def self.main()
        perroquet()
    end
end
Exo6.perroquet()

File: exo6/exo6.py
class exo6:
    @staticmethod
    def perroquet():
        ipt = raw_input("Veuillez ecrire quelque chose\n")
        print("Vous avez ecrit : " + ipt + "\n")
        
        actionValide = False
        while(actionValide == False):
            action = raw_input("Voulez vous quitter ? o/n : ")
            if (action == "o" or action == "O"):
                return
            elif (action == "n" or action == "N"):
                actionValide = True
            else:
                print("Veuillez entrer une reponse valide\n")
        
        print("\n");
        exo6.perroquet()
    
    @staticmethod
    def main():
        exo6.perroquet()

exo6.perroquet()